### PR TITLE
Ask user to accept the electronic form on claiming tokens

### PR DESCRIPTION
**Pull request by 4p00rv**

### CODE REVIEW

File: src/javascript/binary/websocket_pages/user/ico_claim_form.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        };
+        BinarySocket.send(ico_req);
+
+        BinarySocket.wait('get_settings', 'ico_status').then(() => {
+            const first_name     = toTitleCase(State.getResponse('get_settings.first_name'));
+            const last_name      = toTitleCase(State.getResponse('get_settings.last_name'));
+            const full_name      = `${first_name} ${last_name}`;
+            const decimal_places = getDecimalPlaces(currency);
+            const final_price    = (State.getResponse('ico_status.final_price')).toFixed(decimal_places);
+            const date_today     = window.time.format('DD MMM YYYY');
```
Comment (negarn) [inline src/javascript/binary/websocket_pages/user/ico_claim_form.js]: I think we should handle the case if window.time hasn't been initialized

File: src/javascript/binary/websocket_pages/user/ico_portfolio.js [hunk 2]
```diff
@@ -1,8 +1,10 @@
 const Portfolio        = require('./account/portfolio').Portfolio;
 const ViewPopup        = require('./view_popup/view_popup');
 const BinarySocket     = require('../socket');
+const loadUrl       = require('../../base/binary_pjax').load;
```
Comment (negarn) [inline src/javascript/binary/websocket_pages/user/ico_portfolio.js]: equal sign isn't aligned

File: src/javascript/binary/websocket_pages/user/ico_portfolio.js [hunk 3]
```diff
@@ -35,7 +37,7 @@ const ICOPortfolio = (() => {
         let status_text = 'Ended';
         if (/unsuccessful/i.test(long_code)) {
             status_text = 'Refund Bid';
-        } else if (/successful/i.test(long_code)) {
+        } else if (/ successful/i.test(long_code)) {
```
Comment (negarn) [inline src/javascript/binary/websocket_pages/user/ico_portfolio.js]: will long code always be in english?

File: src/javascript/binary/websocket_pages/user/ico_portfolio.js [hunk 4]
```diff
@@ -98,11 +100,27 @@ const ICOPortfolio = (() => {
...
+                });
+
+            $('a[action="claim"]:not(.button-disabled)')
+                .off('click')
+                .on('click', (e) => {
+                    e.preventDefault();
+                    const url = urlFor('user/ico-claim-form');
+                    // set the contract id for ico-claim-form page.
+                    const contract_id = $(e.target).parent().attr('contract_id');
+                    const tokens      = $(e.target).parent().attr('tokens');
```
Comment (negarn) [inline src/javascript/binary/websocket_pages/user/ico_portfolio.js]: we can store `$(e.target).parent()` in a variable so we only retrieve it once

File: src/templates/user/ico_claim_form.html.tt [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    <li>[% l('the Subscriber holds an account on the Binary.com website, with an account identifier that shall be identified as the initial holder of the Tokens, and that Subscriber has satisfactorily completed KYC (Know Your Client), AML (Anti-Money Laundering) procedures, and has accurately updated his or her name, address, country of residence, and ID details in such account;') %]</li>
+    <li>[% l('the Subscriber understands that the Tokens are a risky investment, that he/she is investing entirely at his/her own risk, and may lose the entirety of his/her investment; and') %]</li>
+    <li>[% l('the Tokens will be issued on, and the Subscriber will be bound by, the Instrument and the Conditions, including the representations made herein') %]</li>
+</ol>
+<p>[% l('Electronically signed by the Subscriber/for and on behalf of the Subscriber') %]</p>
+<p>
+    <input type="text" disabled="disabled" class="center-text claimer_name" id="name" placeholder="[% l('Name') %]">
+</p>
+<p>[% l('Subscriber') %]</p>
+<p>[% l('Date: ') %] <input type="text" disabled="disabled" class="center-text" id="date_today" placeholder="[% l('Date') %]"></p>
```
Comment (binary-hamedanchi) [inline src/templates/user/ico_claim_form.html.tt]: Please do not rely on trailing space in translations.
How about `<p>[% l('Date') %]: <input ...` to also avoid multiple translations for one word?

File: src/javascript/binary/websocket_pages/user/ico_portfolio.js (hunk 3 above)
Comment (4p00rv) [reply src/javascript/binary/websocket_pages/user/ico_portfolio.js]: @negarn No, but the backend doesn't provide any other information to determine the bid status.

File: src/javascript/app/pages/user/ico_claim_form.js [hunk 6]
```diff
@@ -0,0 +1,60 @@
+const moment           = require('moment');
+const BinarySocket     = require('../../base/socket');
+const loadPreviousUrl  = require('../../base/binary_pjax').loadPreviousUrl;
+const Client           = require('../../base/client');
+const State            = require('../../../_common/storage').State;
+const getDecimalPlaces = require('../../common/currency').getDecimalPlaces;
+const toTitleCase      = require('../../../_common/string_util').toTitleCase;
```
Comment (negarn) [inline src/javascript/app/pages/user/ico_claim_form.js]: ordering please :)

File: src/javascript/app/pages/user/ico_claim_form.js [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+            currency,
+        };
+        BinarySocket.send(ico_req);
+
+        BinarySocket.wait('get_settings', 'ico_status').then(() => {
+            const first_name     = toTitleCase(State.getResponse('get_settings.first_name'));
+            const last_name      = toTitleCase(State.getResponse('get_settings.last_name'));
+            const full_name      = `${first_name} ${last_name}`;
+            const decimal_places = getDecimalPlaces(currency);
+            const final_price    = (State.getResponse('ico_status.final_price')).toFixed(decimal_places);
```
Comment (negarn) [inline src/javascript/app/pages/user/ico_claim_form.js]: do we need to handle the case where `State.getResponse('ico_status.final_price')` is null?

File: src/javascript/app/pages/user/ico_claim_form.js [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+            const date_today     = (window.time || moment()).format('DD MMM YYYY');
```
Comment (negarn) [inline src/javascript/app/pages/user/ico_claim_form.js]: is this local time or server time? if it's server time it should be `moment.utc()`

File: src/javascript/app/pages/user/ico_claim_form.js [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+            $('.claimer_name').val(full_name);
+            $('#final_price').val(final_price);
+            $('#date_today').val(date_today);
+            $('#token_count').val(tokens);
+
+            // Add event listeners
+            $('.button')
+                .off('click')
+                .on('click', () => {
+                    // Make some api call?
```
Comment (negarn) [inline src/javascript/app/pages/user/ico_claim_form.js]: is this part pending back-end? if so can we change the comment to a `TODO`?

File: src/javascript/app/pages/user/ico_portfolio.js [hunk 10]
```diff
@@ -1,6 +1,8 @@
 const Portfolio        = require('./account/portfolio/portfolio').Portfolio;
 const ViewPopup        = require('./view_popup/view_popup');
 const BinarySocket     = require('../../base/socket');
+const urlFor           = require('../../../_common/url').urlFor;
+const loadUrl          = require('../../base/binary_pjax').load;
```
Comment (negarn) [inline src/javascript/app/pages/user/ico_portfolio.js]: ordering :)

File: src/sass/app/ico_claim_form.scss [hunk 11]
```diff
@@ -0,0 +1,13 @@
+.ico-claim-form-content {
+    p {
+        font-size: $FONT_SIZE_S;
+    }
+    ol li {
```
Comment (negarn) [inline src/sass/app/ico_claim_form.scss]: can we combine `p, ol li` styles?

File: src/sass/static/pages.scss [hunk 12]
```diff
@@ -302,6 +302,20 @@ body #not_authenticated_financial ul.checked > li {
...
+    }
+    ol li {
+        font-size: $FONT_SIZE_S;
+    }
+    input[disabled='disabled'] {
+        background: $COLOR_LIGHT_GRAY;
+        font-weight: 900;
+        width: 200px;
+    }
+}
```
Comment (negarn) [inline src/sass/static/pages.scss]: why are these duplicated from `src/sass/app/ico_claim_form.scss`?

File: src/templates/app/user/ico_claim_form.html.tt [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+    <li>[% l('the Subscriber has read and understood the Information Memorandum of the Company dated 15 November 2017 and in particular has read and understood the Risk Factors detailed therein;') %]</li>
+    <li>[% l('the Subscriber has observed the laws of all relevant jurisdictions, obtained any requisite governmental exchange controls or other consents, complied with all relevant formalities and paid any issue, transfer or other taxes due in connection with its participation in the Auction in any applicable territory and that it has not taken any action which will or may result in the Issuer being in breach of the legal or regulatory requirements of any jurisdiction;') %]</li>
+    <li>[% l('the Subscriber holds an account on the Binary.com website, with an account identifier that shall be identified as the initial holder of the Tokens, and that Subscriber has satisfactorily completed KYC (Know Your Client), AML (Anti-Money Laundering) procedures, and has accurately updated his or her name, address, country of residence, and ID details in such account;') %]</li>
```
Comment (negarn) [inline src/templates/app/user/ico_claim_form.html.tt]: please change `Binary.com` with placeholder `[_1]` and use `website_name` as the variable

File: src/javascript/app/pages/user/ico_claim_form.js [hunk 14]
```diff
@@ -0,0 +1,60 @@
...
+            $('.claimer_name').val(full_name);
+            $('#final_price').val(final_price);
+            $('#date_today').val(date_today);
+            $('#token_count').val(tokens);
+
+            // Add event listeners
+            $('.button')
+                .off('click')
+                .on('click', () => {
+                    //TODO add api call for claim token.
```
Comment (negarn) [inline src/javascript/app/pages/user/ico_claim_form.js]: missing space after `//`, travis test failing

**Pull request merged**